janusgraph-lucene/src/main/java/org/janusgraph/diskstorage/lucene/LuceneIndex.java

```diff
@@ -596,43 +666,43 @@ private SearchParams convertQuery(Condition<?> condition, final KeyInformation.S
-                            final BooleanQuery.Builder q = new BooleanQuery.Builder();
-                            q.add(new MatchAllDocsQuery(), BooleanClause.Occur.MUST);
-                            q.add(new TermQuery(new Term(key, (String) value)), BooleanClause.Occur.MUST_NOT);
-                            params.addQuery(q.build());
-                        } else if (janusgraphPredicate == Text.FUZZY) {
-                            params.addQuery(new FuzzyQuery(new Term(key, (String) value)));
-                        } else if (janusgraphPredicate == Text.CONTAINS_FUZZY) {
-                            value = ((String) value).toLowerCase();
-                            final Builder b = new BooleanQuery.Builder();
-                            for (final String term : Text.tokenize((String) value)) {
-                                b.add(new FuzzyQuery(new Term(key, term)), BooleanClause.Occur.MUST);
-                            }
-                            params.addQuery(b.build());
-                        } else
-                            throw new IllegalArgumentException("Relation is not supported for string value: " + janusgraphPredicate);
+                    final Mapping map = Mapping.getMapping(information.get(key));
+                    if ((map == Mapping.DEFAULT || map == Mapping.TEXT) && !Text.HAS_CONTAINS.contains(janusgraphPredicate))
+                        throw new IllegalArgumentException("Text mapped string values only support CONTAINS queries and not: " + janusgraphPredicate);
+                    if (map == Mapping.STRING && Text.HAS_CONTAINS.contains(janusgraphPredicate))
+                        throw new IllegalArgumentException("String mapped string values do not support CONTAINS queries: " + janusgraphPredicate);
+                    if (janusgraphPredicate == Text.CONTAINS) {
+                        tokenize(params, map, delegatingAnalyzer, ((String) value).toLowerCase(), key, janusgraphPredicate);
+                    } else if (janusgraphPredicate == Text.CONTAINS_PREFIX) {
+                        tokenize(params, map, delegatingAnalyzer, (String) value, key, janusgraphPredicate);
+                    } else if (janusgraphPredicate == Text.PREFIX) {
+                        params.addQuery(new PrefixQuery(new Term(key, (String) value)));
+                    } else if (janusgraphPredicate == Text.REGEX) {
+                        final RegexpQuery rq = new RegexpQuery(new Term(key, (String) value));
+                        params.addQuery(rq);
+                    } else if (janusgraphPredicate == Text.CONTAINS_REGEX) {
+                        // This is terrible -- there is probably a better way
+                        // putting this to lowercase because Text search is supposed to be case insensitive
+                        final RegexpQuery rq = new RegexpQuery(new Term(key, ".*" + (((String) value).toLowerCase()) + ".*"));
```

Comment:
starting the query with .* will give extremely slow RegexpQueries

Reply:
sure. But it is not my code. It is a legacy from the previous implementation (which has issues).

It might be possible to fix, but I haven't spent much time thinking about it.
Lucene uses automaton to convert regex queries into things able to look into an index.
Maybee there is a better solution compiling the regex into such an automaton.

But Regex contained in a text, like all queries beginning with a * drop performance a lot as you can't use prefixes to quickly find the tokens you are interested in, you have to traverse the whole token trie.

My take : 
RegexQuery are experimental in lucene, this demands some investigation (and a lot of time, automaton are hard)
(maybee in a later pr). I'm not sure that this issue can be fixed with the lucene option of allowing queries to begin with a wildcard